### PR TITLE
Kakapo integration

### PR DESCRIPTION
**Pull request by zzarcon**
@MP0w @oskarcieslik just a WIP but check it out

### CODE REVIEW

**Comment by zzarcon** [comment]
@MP0w please review, I'd like to merge it and later each platform starts writing docs

**Comment by MP0w** [comment]
Ok will do it now

File: _includes/github_ribbon.html [hunk 1]
```diff
@@ -0,0 +1 @@
...9.0 C113.8,99.7 119.0,89.6 119.0,89.6 C122.0,82.7 120.5,78.6 120.5,78.6 C119.2,72.0 123.4,76.3 123.4,76.3 C127.3,80.9 125.5,87.3 125.5,87.3 C122.9,97.6 130.6,101.9 134.4,103.2" fill="currentColor" style="transform-origin: 130px 106px;" class="octo-arm"></path><path d="M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z" fill="currentColor" class="octo-body"></path></svg></a><style>.github-corner:hover .octo-arm{animation:octocat-wave 560ms ease-in-out}@keyframes octocat-wave{0%,100%{transform:rotate(0)}20%,60%{transform:rotate(-25deg)}40%,80%{transform:rotate(10deg)}}@media (max-width:500px){.github-corner:hover .octo-arm{animation:none}.github-corner .octo-arm{animation:octocat-wave 560ms ease-in-out}}</style>
```
Comment (MP0w) [inline _includes/github_ribbon.html]: This should use a variable like project_url otherwise can't be reused

Comment (zzarcon) [inline _includes/github_ribbon.html]: Good point, where should I place that variable?

Comment (MP0w) [inline _includes/github_ribbon.html]: In the tank from matter of the page (like kakapo-js)

File: _layouts/kakapo-js.html [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<!DOCTYPE html>
+<html>
+  {% include head.html %}
+  <body>
+    {% include github_ribbon.html %}
+    <section id="banner" class="js-banner">
+      <h1>
+      Kakapo.js
```
Comment (MP0w) [inline _layouts/kakapo-js.html]: The layout should not be specific, get variables from the front matter like title and description and name it more generically , for example this layout could be documentation.html

File: _layouts/kakapo-swift.html [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<!DOCTYPE html>
+<html>
```
Comment (MP0w) [inline _layouts/kakapo-swift.html]: The. You can get rid of this file

File: css/main.scss [hunk 4]
```diff
@@ -49,5 +47,6 @@ $on-laptop:        800px;
 @import
         "base",
         "layout",
-        "syntax-highlighting"
+        "syntax-highlighting",
+        "kakapo"
```
Comment (MP0w) [inline css/main.scss]: maybe should not be always imported since it's not used by other pages and is specific to documentation

Comment (MP0w) [inline css/main.scss]: also generic name would be better, like `documentation`since we will use it for other projects hopefully

Comment (zzarcon) [inline css/main.scss]: Good point, will change it

File: _sass/kakapo.scss [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+    }
+    .arrow{
+      width: 25px;
+      height: 30px;
+      margin-top: 15px;
+      transition: transform .2s;  
+    }
+  }
+  &.js-banner:after{
+    background-image: url('/assets/images/js-bg.jpg');
```
Comment (MP0w) [inline _sass/kakapo.scss]: can we take this out, put it in the layout (documentation) and use a variable for the image name?

Comment (zzarcon) [inline _sass/kakapo.scss]: We just need to use a variable for the className

Comment (MP0w) [inline _sass/kakapo.scss]: But then for every project you need a new class... We can also have variables in the css theoretically if we can't move this out

File: kakapo-js.md [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+const headers = new Headers();
+headers.append('X-auth-token', 'foo-bar');
+const request = new Request('/users/10/friends/55');
+
+fetch(request, {method: 'POST', headers}).then(response => {
+  console.log(response.message);
+});
+```  
+
+### Fetch & XMLHttpRequest support
```
Comment (zzarcon) [inline kakapo-js.md]: @MP0w I'm struggling with this... I can't manage to make this look fine... For some reason is not being parsed and is just outputted like text in the page. Any idea?

Comment (groyoh) [inline kakapo-js.md]: May come from `&`?

Comment (MP0w) [inline kakapo-js.md]: Or try a double space after the triple ticks

**Pull request merged**